core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -1682,7 +1682,14 @@ private long createDirectoryAndJournal(LockedInodePath inodePath, CreateDirector
       CreateDirectoryOptions options) throws InvalidPathException, FileAlreadyExistsException,
       IOException, AccessControlException, FileDoesNotExistException {
     try {
-      return mInodeTree.createPath(inodePath, options);
+      InodeTree.CreatePathResult createResult = mInodeTree.createPath(inodePath, options);
+      InodeDirectory inodeDirectory = (InodeDirectory) inodePath.getInode();
+      // If inodeDirectory's ttl not equals Constants.NO_TTL, it should insert into mTtlBuckets
+      if (createResult.getCreated().size() > 0) {
```

Comment:
I think we also need to add this logic to journal replay (see the InodeFileEntry if branch in the processJournalEntry) ... there should be a unit test for this as well